sdk/iot/Azure.Iot.Hub.Service/src/DevicesClient.cs

```diff
@@ -13,13 +17,489 @@ namespace Azure.Iot.Hub.Service
+        /// where device and twin have a definition, the device value will override the twin value.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The result of the bulk operation.</returns>
+        public virtual Response<BulkRegistryOperationResult> CreateIdentitiesWithTwin(IDictionary<DeviceIdentity, TwinData> devices, CancellationToken cancellationToken = default)
+        {
+            IEnumerable<ExportImportDevice> registryOperations = devices
+                .ToList()
+                .Select(x => new ExportImportDevice()
+                {
+                    Id = x.Key.DeviceId,
+                    Authentication = x.Key.Authentication,
+                    Capabilities = x.Key.Capabilities,
+                    DeviceScope = x.Key.DeviceScope,
+                    ParentScopes = x.Key.ParentScopes,
+                    Status = string.Equals(ExportImportDeviceStatus.Disabled.ToString(), x.Key.Status?.ToString(), StringComparison.OrdinalIgnoreCase) ? ExportImportDeviceStatus.Disabled : ExportImportDeviceStatus.Enabled,
+                    StatusReason = x.Key.StatusReason,
+                    Tags = x.Value.Tags,
+                    Properties = new PropertyContainer(x.Value.Properties?.Desired, x.Value.Properties?.Reported),
+                    ImportMode = ExportImportDeviceImportMode.Create
+                });
+
+            return _registryManagerClient.BulkDeviceCrud(registryOperations, cancellationToken);
+        }
+
+        /// <summary>
+        /// Create multiple devices. A maximum of 100 creations can be done per call, and each device identity must be unique. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
+        /// </summary>
+        /// <param name="deviceIdentities">The devices to create.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The result of the bulk operation.</returns>
+        public virtual async Task<Response<BulkRegistryOperationResult>> CreateIdentitiesAsync(IEnumerable<DeviceIdentity> deviceIdentities, CancellationToken cancellationToken = default)
+        {
+            IEnumerable<ExportImportDevice> registryOperations = deviceIdentities
+                .ToList()
+                .Select(x => new ExportImportDevice()
+                {
+                    Id = x.DeviceId,
+                    Authentication = x.Authentication,
+                    Capabilities = x.Capabilities,
+                    DeviceScope = x.DeviceScope,
+                    ParentScopes = x.ParentScopes,
+                    Status = string.Equals(ExportImportDeviceStatus.Disabled.ToString(), x.Status?.ToString(), StringComparison.OrdinalIgnoreCase) ? ExportImportDeviceStatus.Disabled : ExportImportDeviceStatus.Enabled,
```

Comment:
nit: break this line into multiple lines, same for the rest off them.

Reply:
This is admittedly an ugly line of code, but where would you insert newlines?

Reply:
We usually follow this pattern:
string.Equals(
    ExportImportDeviceStatus.Disabled.ToString(),
    x.Status?.ToString(),
    StringComparison.OrdinalIgnoreCase) ? ExportImportDeviceStatus.Disabled : ExportImportDeviceStatus.Enabled

Reply:
tab for each parameter when there are many

Reply:
```
Status = string.Equals(ExportImportDeviceStatus.Disabled.ToString(), x.Status?.ToString(), StringComparison.OrdinalIgnoreCase) 
              ? ExportImportDeviceStatus.Disabled 
              : ExportImportDeviceStatus.Enabled
```